### PR TITLE
Fixes AddEffWhenHit not triggering Magic or Misc

### LINKED ISSUE #5623: bonus3 bAddEffWhenHit trigger condition ATF_MAGIC doesn't work, it won't be activated by magic attack.
rAthena Hash: https://github.com/rathena/rathena/commit/d065de4ba0e0d71c0231343c647d0118bced4ba2

Client Date: 20200304

Server Mode: Renewal

Description of Issue:

bonus3 bAddEffWhenHit trigger condition ATF_MAGIC doesn't work, it won't be activated by magic attack.

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: Fixes #5623

* **Server Mode**: Pre-renewal and Renewal

* **Description of Pull Request**: 
  * Fixes an issue with AddEffWhenHit not triggering on Magic or Misc type attacks.
Thanks to @24msz6eo!

### CODE REVIEW

File: src/map/skill.cpp [hunk 1]
```diff
@@ -2389,24 +2389,23 @@ int skill_counter_additional_effect (struct block_list* src, struct block_list *
 	sd = BL_CAST(BL_PC, src);
 	dstsd = BL_CAST(BL_PC, bl);
 
-	if(dstsd && attack_type&BF_WEAPON) {	//Counter effects.
-		enum sc_type type;
-		unsigned int time;
-
+	if(dstsd && attack_type & (BF_WEAPON | BF_MAGIC | BF_MISC)) {	//Counter effects.
```
Comment (Atemo) [inline src/map/skill.cpp:2392]:
```suggestion
	if(dstsd && attack_type & BF_WEAPONMASK) {	//Counter effects.
```

File: src/map/skill.cpp [hunk 2]
```diff
@@ -2389,24 +2389,23 @@ int skill_counter_additional_effect (struct block_list* src, struct block_list *
...
 				continue;
-
 			if ((it.flag&(ATF_LONG|ATF_SHORT)) != (ATF_LONG|ATF_SHORT)) {	//Trigger has range consideration.
 				if((it.flag&ATF_LONG && !(attack_type&BF_LONG)) ||
 					(it.flag&ATF_SHORT && !(attack_type&BF_SHORT)))
 					continue; //Range Failed.
 			}
-			type = it.sc;
-			time = it.duration;
+			if (!(it.flag & attack_type))
```
Comment (Atemo) [inline src/map/skill.cpp:2404]: Type/value differents (`attack_type` : `BF_...`) (`it.flag` : `ATF_...`)

**Review by Atemo — CHANGES_REQUESTED** [review]

**Comment by 24msz6eo** [comment]
bonus3 bAddEffWhenHit,Eff_Freeze,10000,ATF_WEAPON;

The above script will be effective for magic attacks

**Comment by 24msz6eo** [comment]
This pr needs more judgment, for example, Garm Card, will become physical and magical attacks will be effective

**Pull request merged**